### PR TITLE
feat(data-handlers): add X-Cache header

### PR DESCRIPTION
**Pull request by karlprieb**

### CODE REVIEW

File: src/data/read-through-data-cache.ts [hunk 1]
```diff
@@ -132,13 +133,15 @@ export class ReadThroughDataCache implements ContiguousDataSource {
...
     );
+
     if (cacheData !== undefined) {
       return {
         hash: attributes?.hash,
         stream: cacheData.stream,
         size: cacheData.size,
-        sourceContentType: dataAttributes?.contentType,
-        verified: dataAttributes?.verified ?? false,
+        sourceContentType: attributes?.contentType,
```
Comment (karlprieb) [inline src/data/read-through-data-cache.ts:142]: @djwhitt I believe the intent was to use `attributes` instead of `dataAttributes` here, right?

**Comment by karlprieb** [comment]
@djwhitt should we set `const size = parentData.size ?? dataSize;` as we're returning the parent data here?
https://github.com/ar-io/ar-io-node/blob/169d2c7551dc9f86a8a5b7e4406e06482c73f5c6/src/data/read-through-data-cache.ts#L105-L106

File: src/data/read-through-data-cache.ts (hunk 1 above)
Comment (djwhitt) [reply src/data/read-through-data-cache.ts:142]: Yup. Good catch. This looks like a mistake.

**Comment by djwhitt** [comment]
> @djwhitt should we set `const size = parentData.size ?? dataSize;` as we're returning the parent data here?
> 
> https://github.com/ar-io/ar-io-node/blob/169d2c7551dc9f86a8a5b7e4406e06482c73f5c6/src/data/read-through-data-cache.ts#L105-L106

This one is intentional. I think in theory either will work, but the goal is to treat the `dataAttributes` as authoritative.

**Pull request merged**